### PR TITLE
Fix sign in button is not aligned

### LINKED ISSUE #2226: "Sign in" is not aligned
Screen Shot 2020-12-19 at 9 37 27 AM

It happens when you're signed in too. In all browsers.

### PR DESCRIPTION
**Pull request by tannerdolby**
Fixes #2226 

@peterbe Ready for review. I left some things commented out until I got the approval to remove them. 

Also, the comments in `../login/index.scss` for the min/max media condition I could remove, I forgot to elaborate more so its probably not very helpful in its current state. The 768px value could also be converted to 48em to be more uniform.

### CODE REVIEW

**Comment by peterbe** [comment]
Note-to-self; See https://github.com/mdn/yari/issues/2226#issuecomment-749400424 for more context.

**Comment by peterbe** [comment]
> I left some things commented out until I got the approval to remove them.

If your PR isn't ready for merging, you can create a "Draft PR" and those can be reviewed too. It's quite handy when you don't want to go that "last extra mile" (e.g. tests or clean-up) until peers have confirmed that you're on the right track. 

But I think @schalkneethling can review it now and he might have to do the "Request changes" as a reminder for you to clean up the commented out stuff.

**Comment by tannerdolby** [comment]
Got it, I will keep that in mind and create a draft PR next time.

**Comment by peterbe** [comment]
@tannerdolby Can you merge in the latest `master` into this branch and push?

**Review by peterbe — CHANGES_REQUESTED** [review]
The center of the avatar doesn't appear to line up. 
Screen Shot 2021-01-07 at 4 23 53 PM

But, when you're not signed in it's aligned:
Screen Shot 2021-01-07 at 4 25 47 PM

File: client/src/ui/organisms/header/index.scss [hunk 1]
```diff
@@ -102,7 +102,7 @@
     }
 
     .auth-container {
-      align-self: center;
+      // align-self: center;
```
Comment (peterbe) [inline client/src/ui/organisms/header/index.scss:105]: left-over debugging?

File: client/src/ui/molecules/login/index.scss [hunk 2]
```diff
@@ -4,8 +4,14 @@
 .auth-container {
   margin-top: $base-unit * 2;
 
-  @media #{$mq-large-desktop-and-up} {
-    margin-top: 0;
+  /* This media condition could be removed as margin-top isn't changing from above */
+  // @media #{$mq-large-desktop-and-up} {
+  //   margin-top: 0;
+  // }
```
Comment (peterbe) [inline client/src/ui/molecules/login/index.scss:10]: Is this debugging?

**Comment by tannerdolby** [comment]
@peterbe Yeah, sure thing. I can merge with `master` in this branch and push.

**Comment by peterbe** [comment]
By the way, to simulate being signed in, this is what I use:
```
SERVER_FAKE_V1_API=true
```
...in my `.env` file. 
And this file:
```
▶ cat fake-v1-api/whoami.json
{
  "username": "peter",
  "is_authenticated": true,
  "avatar_url": "https://www.peterbe.com/avatar.random.png",
  "email": "peter@example.com",
  "subscriber_number": 2,
  "is_subscriber": true,
  "wiki_contributions": 0,
  "waffle": {
    "flags": {
      "kumaediting": true,
      "section_edit": true,
      "spam_checks_enabled": true,
      "subscription": true,
      "subscription_banner": true,
      "subscription_form": true,
      "user_interviews": true,
      "developer_needs": false
    },
    "switches": {},
    "samples": {}
  }
}
```

Does that help?

File: client/src/ui/molecules/login/index.scss (hunk 2 above)
Comment (tannerdolby) [reply client/src/ui/molecules/login/index.scss:10]: This was just a comment I left as a note to whoever reviewed before completely removing it. I will remove lines 7-10 in my next push.

File: client/src/ui/organisms/header/index.scss (hunk 1 above)
Comment (tannerdolby) [reply client/src/ui/organisms/header/index.scss:105]: Ya, this is what was causing the lowered alignment for the "Sign in" button. I will remove it with my next push.

**Comment by tannerdolby** [comment]
Yes this definitely helps. I will simulate a sign in and see what I can do to fix the mis-aligned avatar.

I was going to ask how to simulate signing in, so thank you! I followed the trail of files where the "Sign in" permalink and functionality was being created but never actually figured out how to simulate it and ended up just editing the HTML to put the "Sign in" text into the empty anchor tag.

**Comment by peterbe** [comment]
For the record, sign in only works in two ways:
1) Simulate with that `SERVER_FAKE_V1_API` trick
2) Have a local version of Kuma running and point to it with `REACT_APP_KUMA_HOST`

Using the fake thing with mock `.json` files is certainly easier :)

**Comment by tannerdolby** [comment]
Found the fix for the avatar being mis-aligned. Once again it was the `align-self: center` declaration here:

Screen Shot 2021-01-07 at 11 07 31 PM

```css
@media (min-width: 47.9385em)
.page-header-main .auth-container {
    align-self: center;
    grid-column: 2/3;
    grid-row: 2/3;
    justify-self: flex-end;
}
```

After removing `align-self: center` from the above media query, the avatar logo is aligned correctly. 

Screen Shot 2021-01-07 at 11 07 20 PM

**Comment by tannerdolby** [comment]
Shucks, I forgot to sign the merge with `master` commit since it was done in vim, I will try my hand at a rebase to drop the commit and re-commit with a signed one.

File: client/src/ui/molecules/login/index.scss [hunk 3]
```diff
@@ -4,8 +4,14 @@
...
+
+  /* Handling < 1199px Sign in button alignment, if there is a $var then use that instead */
+  @media (max-width: 74.9375em) and (min-width: 768px) {
+    margin-top: $base-unit * 3;
   }
```
Comment (schalkneethling) [inline client/src/ui/molecules/login/index.scss:15]: We should not introduce new media queries. We have defined media queries to use for this purpose. We should always follow a mobile-first approach.

These are the media queries to use:

```
$mq-large-desktop-and-up: "all and (min-width: #{$small-desktop-ends})";
$mq-small-desktop-and-up: "all and (min-width: #{$small-desktop-starts})";
$mq-tablet-and-up: "all and (min-width: #{$tablet-starts})";
```

For example:

```
@media #{$mq-tablet-and-up} {
   ...
}
```

**Review by schalkneethling — CHANGES_REQUESTED** [review]

File: client/src/ui/molecules/login/index.scss (hunk 3 above)
Comment (tannerdolby) [reply client/src/ui/molecules/login/index.scss:15]: @schalkneethling Got it! I will make sure to update this.

**Comment by schalkneethling** [comment]
@tannerdolby Thank you for this PR. I am rethinking the header a bit and so, let's not worry about tablet for this particular issue. On mobile we are fine so, in this PR, let's focus on fixing the desktop layout. For that, I would suggest the following:

Remove the `align-self: center;` as you have done. The on `.page-header-main` adjust it to the following:

```
.page-header-main {
  display: grid;
  align-items: center;
}
```

then also remove the following `padding`:

```
.dropdown-container .dropdown-menu-label {
  padding: 0 6px 6px;
}
```

That gives us the following:

### Signed out

Screenshot 2021-01-08 at 09 56 56

### Signed in

Screenshot 2021-01-08 at 09 58 30

**Comment by tannerdolby** [comment]
@schalkneethling Your welcome. Okay sounds good, I removed that media query for tablet styling you mentioned in the requested changes.

I've also removed the `align-self: center` spots, and updated `.page-header-main` to include `align-items: center`. For removing the `padding` from `.dropdown-container .dropdown-menu-label`, I noticed `padding` was the only property within that style block therefore I completely removed 

```css
@media #{$mq-tablet-and-up} {
  padding: 0 $base-unit $base-unit;
}
```

**Comment by schalkneethling** [comment]
> @schalkneethling Your welcome. Okay sounds good, I removed that media query for tablet styling you mentioned in the requested changes.
> 
> I've also removed the `align-self: center` spots, and updated `.page-header-main` to include `align-items: center`. For removing the `padding` from `.dropdown-container .dropdown-menu-label`, I noticed `padding` was the only property within that style block therefore I completely removed
> 
> ```css
> @media #{$mq-tablet-and-up} {
>   padding: 0 $base-unit $base-unit;
> }
> ```

I guess you still need to push your latest commit?

**Comment by tannerdolby** [comment]
Yeah, I'm just about to push my commits (was doing some last minute checks)

**Comment by tannerdolby** [comment]
Pushed!

**Review by schalkneethling — APPROVED** [review]

**Review by peterbe — APPROVED** [review]

**Pull request merged**